src/main/java/com/github/fishio/audio/Sound.java

```diff
@@ -0,0 +1,158 @@
+	 * 		If an IOException occurs while reading the given input stream.
+	 * @throws UnsupportedAudioFileException
+	 * 		If the audio format is unsupported.
+	 */
+	protected void readAudioStream(InputStream inputStream) throws IOException, UnsupportedAudioFileException {
+		try (AudioInputStream ais = AudioSystem.getAudioInputStream(inputStream)) {
+			//Load the format
+			this.format = ais.getFormat();
+			
+			//Read the bytes from the audio file.
+			ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+	
+			int nRead;
+			byte[] data = new byte[READ_BUFFER_SIZE];
+	
+			while ((nRead = ais.read(data, 0, data.length)) != -1) {
+			  buffer.write(data, 0, nRead);
+			}
+	
+			buffer.flush();
+	
+			this.contents = buffer.toByteArray();
+		}
+	}
+	
+	/**
+	 * Read an input stream in mp3 format.
+	 * 
+	 * @param inputStream
+	 * 		the stream to read from.
+	 * 
+	 * @throws IOException
+	 * 		If an IOException occurs while reading the given input stream.
+	 * @throws UnsupportedAudioFileException
+	 * 		If the audio format is unsupported.
+	 */
+	protected void readMp3Stream(InputStream inputStream) throws IOException, UnsupportedAudioFileException {
+		try (AudioInputStream ais = AudioSystem.getAudioInputStream(inputStream)) {
+			//We want the file in the following format, so we can play it.
+			this.format = new AudioFormat(AudioFormat.Encoding.PCM_SIGNED, 
+					ais.getFormat().getSampleRate(),
+                    16,
+                    ais.getFormat().getChannels(),
+                    ais.getFormat().getChannels() * 2,
+                    ais.getFormat().getSampleRate(),
+                    false);
+			
+			try (AudioInputStream ais2 = AudioSystem.getAudioInputStream(this.format, ais)) {
+				//Read the bytes from the audio file.
+				ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+				int nRead;
+				byte[] data = new byte[READ_BUFFER_SIZE];
+
+				try {
+					while ((nRead = ais2.read(data, 0, data.length)) != -1) {
+						buffer.write(data, 0, nRead);
+					}
+				} catch (ArrayIndexOutOfBoundsException ex) {
+					//If the MP3 is malformed, we can get an ArrayIndexOutOfBoundsException.
+					Log.getLogger().log(LogLevel.INFO, "[Sound] Malformed MP3");
+				}
+
+				buffer.flush();
+
+				this.contents = buffer.toByteArray();
+			}
+		}
+	}
+	
+	/**
+	 * @return
+	 * 		a new FishClip that hold a clip for the audio fragment 
+	 * 		corresponding to this Sound object.
+	 * 
+	 * @throws LineUnavailableException
+	 * 		See {@link Clip#open()}.
+	 */
+	@SuppressWarnings("resource")
+	public FishClip getClip() throws LineUnavailableException {
+		Clip clip = AudioSystem.getClip();
+		clip.open(format, contents, 0, contents.length);
+		
+		return new FishClip(clip, effect);
+	}
+	
+	/**
+	 * @return
+	 * 		<code>true</code> if this FishSound represents a Sound Effect.<br>
+	 * 		<code>false</code> if it represents (background) music.
+	 */
+	public boolean isEffect() {
+		return effect;
+	}
```

Comment:
Do you really need the same isEffect() method in both the Sound and FishClip classes? Shouldn't the effect boolean of the FishClip class be linked to the effect boolean of the Sound class, as the sound file defines whether the sound is background music or a sound effect? Perhaps I do not really understand the connection between Sound and FishClip correctly.

< organisation